x/stake/keeper/stake.go

```diff
@@ -33,6 +33,72 @@ func (k Keeper) GetStake(ctx sdk.Context, vendorID uint32, postID curatingtypes.
 	return stake, true, nil
 }
 
+// BuyCreatorCoin delegates an amount to a validator and associates a post
+func (k Keeper) PerformBuyCreatorCoin(
+	ctx sdk.Context,
+	username string,
+	creator sdk.AccAddress,
+	buyer sdk.AccAddress,
+	valAddr sdk.ValAddress,
+	amount sdk.Int) error {
+
+	coin := sdk.NewCoin(fmt.Sprintf("%s/%s", username, creator.String()), amount)
+
+	stake, found, err := k.GetStake(ctx, 0, curatingtypes.PostID{}, buyer)
```

Comment:
so this is staking to empty post id(prob 0) and vendor 0 all the times? I guess this is what you meant about better using a forked staking module

Reply:
Yeah, this is just for accounting to keep track of delegations for creator coins since it'll be needed for sells.